### PR TITLE
Eslint prettier

### PR DESCRIPTION
**Pull request by BartDeCaluwe**
This PR will:

- Add eslint and prettier config
- Add linting step to pipeline
- Linting and formatting changes caused by eslint

## Checklist

- [x] I used the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
- [x] Unit tests have been created if a new feature was added.

### CODE REVIEW

**Comment by RuvendeGroote** [comment]
is the prettier stuff being applied? you should add an entry to package.json to specify the prettier configuration to use + a format script and do it as a step in the pipeline

**Comment by BartDeCaluwe** [comment]
> is the prettier stuff being applied? you should add an entry to package.json to specify the prettier configuration to use + a format script and do it as a step in the pipeline

Doesn't the linter do the prettier formatting as well? Or is that just because I specified it that way in my settings.
Ah but that won't mean it will be formatted in the pipeline as well. I'll add it!

**Comment by BouweCeunen** [comment]
Small conflict still, also seems build is failing `
Type error: 'Component' cannot be used as a JSX component.`

**Comment by BartDeCaluwe** [comment]
> Small conflict still, also seems build is failing ` Type error: 'Component' cannot be used as a JSX component.`

Unfortunately the build is failing since last week because of this [issue](https://github.com/vercel/next.js/issues/35986). I'm keeping my eye on that thread to see if any fix comes around.

File: .eslintrc.json [hunk 1]
```diff
@@ -1,8 +1,9 @@
 {
-	"plugins": ["@typescript-eslint"],
-	"extends": ["next/core-web-vitals", "plugin:@typescript-eslint/recommended", "prettier"],
+	"extends": [
+		"@credix/eslint-config",
+		"next"
+	],
 	"rules": {
```
Comment (RuvendeGroote) [inline .eslintrc.json]: this is not necessary anymore

**Review by RuvendeGroote — DISMISSED** [review]

File: .github/workflows/cicd.yaml [hunk 2]
```diff
@@ -28,6 +28,35 @@ jobs:
...
+      - name: Yarn install
+        run: yarn
+
+      # Lint
+      - name: Lint
+        run: yarn lint
+
+      # Format
+      - name: Check formatting
+        run: yarn format
```
Comment (BouweCeunen) [inline .github/workflows/cicd.yaml:49]: I believe linting is already done in the `ci` step a few lines below, maybe just add `yarn format` there: `yarn lint && ...`?

**Review by BouweCeunen — APPROVED** [review]

**Review by RuvendeGroote — APPROVED** [review]

**Pull request merged**